### PR TITLE
add sunspot aurora test machine

### PR DESCRIPTION
**Pull request by xyuan**
Add sunspot aurora test machine to E3SM for both cpu and gpu using oneAPI compiler.

[BFB]

### CODE REVIEW

**Comment by sarats** [comment]
Generally speaking, should we try to have both modes (JIT and AoT) for debugging purposes or just one?

File: components/cmake/build_model.cmake [hunk 1]
```diff
@@ -74,12 +74,17 @@ function(build_model COMP_CLASS COMP_NAME)
...
         # HIP_FLAGS are set through Macros.cmake
         # For instance: cime_config/machines/cmake_macros/crayclanggpu_frontier.cmake
-		set(YAKL_HIP_FLAGS "${CPPDEFS} ${HIP_FLAGS}")
-    else()
+	set(YAKL_HIP_FLAGS "${CPPDEFS} ${HIP_FLAGS}")
+      elseif (USE_SYCL)
+        set(YAKL_ARCH "SYCL")
+        # SYCL_FLAGS is set through Macros.cmake
+	# For instance: cime_config/machines/cmake_macros/oneapi-ifxgpu_sunspot.cmake
+        set(YAKL_SYCL_FLAGS "${CPPDEFS} ${SYCL_FLAGS}")
```
Comment (sarats) [inline components/cmake/build_model.cmake:86]: I had to add CPPDEFS to YAKL flags for HIP to make passing MPI include paths etc. easier. 
However, Nvidia/CUDA doesn't like that apparently. So, I presume Intel would also be okay with CPPDEFS added?

cc @mrnorman

**Comment by sarats** [comment]
Add performance archiving support to a project-wide location. See other machines for template. 

Example:
```
 606     <SAVE_TIMING_DIR>/lustre/orion/cli115/world-shared/frontier</SAVE_TIMING_DIR>
 607     <SAVE_TIMING_DIR_PROJECTS>.*</SAVE_TIMING_DIR_PROJECTS>
```

File: cime_config/machines/config_batch.xml [hunk 2]
```diff
@@ -193,6 +193,33 @@
     </directives>
   </batch_system>
 
+  <batch_system type="pbspro" >
```
Comment (sarats) [inline cime_config/machines/config_batch.xml:196]: Could you summarize how this section differs from the existing 'pbs' section above? Almost looks similar.

File: cime_config/machines/config_batch.xml [hunk 3]
```diff
@@ -504,6 +531,14 @@
       </queues>
     </batch_system>
 
+    <batch_system MACH="sunspot" type="pbspro">
+      <batch_submit>qsub</batch_submit>     
+      <queues>
+        <queue walltimemax="00:02:00" jobmin="1" jobmax="128" default="true">workq</queue>
+	<queue walltimemax="00:30:00" jobmin="1" jobmax="128">diags</queue>
```
Comment (sarats) [inline cime_config/machines/config_batch.xml:538]: Minor: indenting.

File: components/cmake/build_model.cmake [hunk 4]
```diff
@@ -74,12 +74,17 @@ function(build_model COMP_CLASS COMP_NAME)
         # CUDA_FLAGS is set through Macros.cmake
         # For instance: cime_config/machines/cmake_macros/gnugpu_summit.cmake
         set(YAKL_CUDA_FLAGS "${CPPDEFS} ${CUDA_FLAGS}")
-	elseif (USE_HIP)
-		set(YAKL_ARCH "HIP")
+      elseif (USE_HIP)
+	set(YAKL_ARCH "HIP")
```
Comment (sarats) [inline components/cmake/build_model.cmake:78]: Minor: Fix indenting.

File: cime_config/machines/config_machines.xml [hunk 5]
```diff
@@ -3162,6 +3162,117 @@
...
+     </environment_variables>
+     <environment_variables mpilib="impi">
+        <env name="I_MPI_DEBUG">10</env>
+        <env name="I_MPI_PIN_DOMAIN">omp</env>
+        <env name="I_MPI_PIN_ORDER">spread</env>
+        <env name="I_MPI_PIN_CELL">unit</env>
+     </environment_variables>
+     <environment_variables compiler="oneapi-ifxgpu">
+        <env name="ONEAPI_DEVICE_SELECTOR">level_zero:gpu</env>
+        <env name="ONEAPI_MPICH_GPU">NO_GPU</env>
```
Comment (sarats) [inline cime_config/machines/config_machines.xml:3251]: So, we don't use GPU-aware MPI on Sunspot because of issues, correct? Remind me ,do we have existing tickets open with Intel on those?

File: cime_config/machines/config_machines.xml [hunk 6]
```diff
@@ -3162,6 +3162,117 @@
...
+        <env name="MPIR_CVAR_ENABLE_GPU">0</env>
+        <env name="romio_cb_read">disable</env>
+        <env name="romio_cb_write">disable</env>
+        <env name="SYCL_CACHE_PERSISTENT">1</env>
+        <env name="GATOR_INITIAL_MB">4000MB</env>
```
Comment (sarats) [inline cime_config/machines/config_machines.xml:3256]: @mrnorman Do you want us to set GATOR_INITIAL_MB in E3SM config or handle based on machine in YAKL?

File: cime_config/machines/config_machines.xml (hunk 5 above)
Comment (xyuan) [reply cime_config/machines/config_machines.xml:3251]: yes, we turn it off at the moment

File: cime_config/machines/config_batch.xml (hunk 2 above)
Comment (xyuan) [reply cime_config/machines/config_batch.xml:196]: there are slightly different in the command options, see the difference listed in this webpage, https://docs.alcf.anl.gov/running-jobs/pbs-qsub-options-table/

File: cime_config/machines/config_machines.xml (hunk 5 above)
Comment (xyuan) [reply cime_config/machines/config_machines.xml:3251]: we will eventually have this function on in the production run, when it tested fine on sunspot.

File: cime_config/machines/config_machines.xml (hunk 6 above)
Comment (xyuan) [reply cime_config/machines/config_machines.xml:3256]: > 
It should be fine AoT in debug mode too, as the JIT and AoT build time are comparable at the moment, but the runtime has large differences

File: cime_config/machines/config_batch.xml (hunk 2 above)
Comment (sarats) [reply cime_config/machines/config_batch.xml:196]: It looks like lines 168-194 are the same as 196-221 (except commented out one). What am I missing?

File: cime_config/machines/config_machines.xml (hunk 6 above)
Comment (mrnorman) [reply cime_config/machines/config_machines.xml:3260]: That's up to you. YAKL default to consuming 1/8 of the total device memory. If you need to override that for any reason, you have two options: (1) `GATOR_INITIAL_MB`; or (2) passing a `yakl::InitConfig` object to `yakl::init()` specifying the runtime-determined pool sizes.

Whether or where to set `GATOR_INITIAL_MB`, I'm not sure. YAKL will not be doing anything machine-specific (i.e., no lookups for "frontier" or "summit" or anything in the `dnsdomainname` command. By default, YAKL will just look for the total amount device memory (or host memory for CPU targets) and allocate one eight of it for the initial pool and each additional pool.

Comment (sarats) [reply cime_config/machines/config_machines.xml:3263]: @xyuan Aren't you setting GATOR_INITIAL_MB to less than 1/8th here?
